### ISSUE TITLE
crate fails to install

### ISSUE THREAD

**Issue by pocc** [ostrosco/device_query#40]
When I try to use the program listed in the README and run `cargo build` it fails with this error:

```
 Compiling device_query v0.1.3
error[E0601]: `main` function not found in crate `build_script_build`
  --> /Users/rj/.cargo/registry/src/github.com-1ecc6299db9ec823/device_query-0.1.3/build.rs:1:1
   |
1  | / extern crate pkg_config;
2  | |
3  | | #[cfg(target_os = "windows")]
4  | | fn main() {}
...  |
39 | |     }
40 | | }
   | |_^ consider adding a `main` function to `/Users/rj/.cargo/registry/src/github.com-1ecc6299db9ec823/device_query-0.1.3/build.rs`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0601`.
error: could not compile `device_query`.

To learn more, run the command again with --verbose.
```

I am on Macos, and looking at the file in question, it doesn't look like there's a #[cfg(target_os = "macos")]. 

I am able to build your repo just fine by cloning it and building it a temp directory.

**Comment by ostrosco** [comment]
It looks like you're building 0.1.3. That's a pretty old version. If you can, could you try building 0.2.8 which is the latest version? I haven't been doing releases on Github proper which is probably where the confusion came from. Sorry about that.

**Comment by pocc** [comment]
I think I fixed this by upgrading to Macos 12.0.1 Monterrey. I'll mark this as solved - thank you.
